src/MLJInterface.jl

```diff
@@ -0,0 +1,324 @@
+
+# TODO: Enable `verbosity` being passed to `equation_search`
+function MMI.fit(m::AbstractSRRegressor, verbosity, X, y, w=nothing)
+    return MMI.update(m, verbosity, (; state=nothing), nothing, X, y, w)
+end
+function MMI.update(
+    m::AbstractSRRegressor, verbosity, old_fitresult, old_cache, X, y, w=nothing
+)
+    options = get(old_fitresult, :options, get_options(m))
+    X_t, variable_names, schema = get_matrix_and_schema(X)
+    y_t, y_variable_names, y_schema = format_input_for(m, y)
+    search_state = equation_search(
+        X_t,
+        y_t;
+        niterations=m.niterations,
+        weights=w,
+        variable_names=variable_names,
+        options=options,
+        parallelism=m.parallelism,
+        numprocs=m.numprocs,
+        procs=m.procs,
+        addprocs_function=m.addprocs_function,
+        runtests=m.runtests,
+        saved_state=old_fitresult.state,
+        return_state=true,
+        loss_type=m.loss_type,
+    )
+    fitresult = (;
+        state=search_state,
+        options=options,
+        variable_names=variable_names,
+        schema=schema,
+        y_variable_names=y_variable_names,
+        y_schema=y_schema,
+    )
+    return (fitresult, nothing, full_report(m, fitresult))
+end
+function get_matrix_and_schema(X)
+    sch = MMI.istable(X) ? MMI.schema(X) : nothing
+    Xm_t = MMI.matrix(X; transpose=true)
+    colnames = if sch === nothing
+        [map(i -> "x$(i)", axes(Xm_t, 1))...]
+    else
+        [string.(sch.names)...]
+    end
+    return Xm_t, colnames, sch
+end
+
+function format_input_for(::SRRegressor, y)
+    @assert(
+        !(MMI.istable(y) || (length(size(y)) == 2 && size(y, 2) > 1)),
+        "For multi-output regression, please use `MultitargetSRRegressor`."
+    )
+    return vec(y), nothing, nothing
+end
+function format_input_for(::MultitargetSRRegressor, y)
+    @assert(
+        MMI.istable(y) || (length(size(y)) == 2 && size(y, 2) > 1),
+        "For single-output regression, please use `SRRegressor`."
+    )
+    return get_matrix_and_schema(y)
+end
+
+function MMI.fitted_params(m::AbstractSRRegressor, fitresult)
+    report = full_report(m, fitresult)
+    return (;
+        best_idx=report.best_idx,
+        equations=report.equations,
+        equation_strings=report.equation_strings,
+    )
+end
+function MMI.predict(m::SRRegressor, fitresult, Xnew)
+    params = MMI.fitted_params(m, fitresult)
+    Xnew_t = MMI.matrix(Xnew; transpose=true)
+    eq = params.equations[params.best_idx]
+    out, flag = eval_tree_array(eq, Xnew_t, fitresult.options)
+    !flag && error("Detected a NaN in evaluating expression.")
+    return out
+end
+function MMI.predict(m::MultitargetSRRegressor, fitresult, Xnew)
+    params = MMI.fitted_params(m, fitresult)
+    Xnew_t, variable_names, schema = get_matrix_and_schema(Xnew)
+    @assert(
+        variable_names == fitresult.variable_names,
+        "Variable names do not match fitted regressor."
+    )
```

Comment:
@OkonSamuel Should we also do a check for `schema == fitresult.schema`?

Reply:
No, there isn't any need for that.